js/test/test.js

```diff
@@ -348,78 +397,114 @@ async function runPrivateTests(exchange, symbol) {
 
     const code = getExchangeCode (exchange);
 
-    const balance = await test ('fetchBalance', exchange);
-
-    await test ('fetchAccounts', exchange);
-    await test ('fetchTransactionFees', exchange);
-    await test ('fetchTradingFees', exchange);
-    await test ('fetchStatus', exchange);
-
-    await test ('fetchOrders', exchange, symbol);
-    await test ('fetchOpenOrders', exchange, symbol);
-    await test ('fetchClosedOrders', exchange, symbol);
-    await test ('fetchMyTrades', exchange, symbol);
-    await test ('fetchLeverageTiers', exchange, symbol);
-    await test ('fetchOpenInterestHistory', exchange, symbol);
-
-    await test ('fetchPositions', exchange, symbol);
 
-    if ('fetchLedger' in tests) {
-        await test ('fetchLedger', exchange, code);
+    // if (exchange.extendedTest) {
+
+    //     await test ('InvalidNonce', exchange, symbol);
+    //     await test ('OrderNotFound', exchange, symbol);
+    //     await test ('InvalidOrder', exchange, symbol);
+    //     await test ('InsufficientFunds', exchange, symbol, balance); // danger zone - won't execute with non-empty balance
+    // }
+
+    let tests = {
+        'signIn': [exchange],
+        'fetchBalance': [exchange],
+        'fetchAccounts': [exchange],
+        'fetchTransactionFees': [exchange],
+        'fetchTradingFees': [exchange],
+        'fetchStatus': [exchange],
+        'fetchOrders': [exchange, symbol],
+        'fetchOpenOrders': [exchange, symbol],
+        'fetchClosedOrders': [exchange, symbol],
+        'fetchMyTrades': [exchange, symbol],
+        'fetchLeverageTiers': [exchange, symbol],
+        'fetchOpenInterestHistory': [exchange, symbol],
+        'fetchPositions': [exchange, symbol],
+        'fetchLedger': [exchange, code],
+        'fetchTransactions': [exchange, code],
+        'fetchDeposits': [exchange, code],
+        'fetchWithdrawals': [exchange, code],
+        'fetchBorrowRates': [exchange, code],
+        'fetchBorrowRate': [exchange, code],
+        'fetchBorrowInterest': [exchange, code],
+        'fetchBorrowInterest': [exchange, code, symbol],
+        'addMargin': [exchange, symbol],
+        'reduceMargin': [exchange, symbol],
+        'setMargin': [exchange, symbol],
+        'setMarginMode': [exchange, symbol],
+        'setPositionMode': [exchange, symbol],
+        'setLeverage': [exchange, symbol],
+        'cancelAllOrders': [exchange, symbol],
+        'cancelOrder': [exchange, symbol],
+        'cancelOrders': [exchange, symbol],
+        'fetchCanceledOrders': [exchange, symbol],
+        'fetchClosedOrder': [exchange, symbol],
+        'fetchOpenOrder': [exchange, symbol],
+        'fetchOrder': [exchange, symbol],
+        'fetchOrderTrades': [exchange, symbol],
+        'fetchPosition': [exchange, symbol],
```

Comment:
@ttodua one more question, as you know, some of these methods are swap only (fetchPositions, etc), so I think we should distinguish the symbol used, because it does not make sense to call them with a spot symbol.

We could do something like this:
```Javascript
const swapSymbol = ...
const spotSymbol =  ...
const tests = {
        'fetchOrder': [exchange, spotSymbol],
        'fetchOrderTrades': [exchange, spotSymbol],
        'fetchPosition': [exchange, swapSymbol],
         (...)
}

Reply:
ofc, not only that, but some methods are just for swap 'types' (without symbol at all, like fetchBorrowRateHistories or others I recall). so, we should include those changes in this PR too, right? ok, give me some time.